vault/identity_store_oidc_path_config.go

```diff
@@ -0,0 +1,87 @@
+package vault
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+const (
+	oidcConfigFieldIssuer = "issuer"
+	oidcConfigStorageKey  = "oidc/config/issuer"
+)
+
+type oidcConfig struct {
+	Issuer string `json:"issuer"`
+}
+
+// oidcPathConfig returns the API endpoint for operations on OIDC configuration
+func oidcPathConfig(i *IdentityStore) *framework.Path {
+	return &framework.Path{
+		Pattern: "oidc/config",
+		Fields: map[string]*framework.FieldSchema{
+			oidcConfigFieldIssuer: &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Issuer URL to be used in the iss claim of the token. If not set, Vault's app_addr will be used.",
+			},
+		},
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.CreateOperation: writeOIDCConfig(),
+			logical.ReadOperation:   readOIDCConfig(),
+			logical.UpdateOperation: writeOIDCConfig(),
+		},
+		HelpSynopsis:    "HelpSynopsis here",
+		HelpDescription: "HelpDecription here",
+	}
+}
+
+// unsure if the following methods should be on the identityStore or not
```

Comment:
the convention seems to be having it on the backend struct (in this case, `identityStore`) but not too sure on the why, @kalafut ?

Reply:
It's common to have shared data (caches, locks, etc.) on the backend struct.